File: Kingdom-Fall-UI-Enabled/src/main/java/personnages/Boss.java
package personnages;

import objets.Objet;
import java.util.ArrayList;

public class Boss extends Ennemi {

    private final int ptsArmure;
    private final int donjonDebloque;
    private final String description;

    public Boss (String nom, String description, int niveau, int ptsVie, int att, int xpDrop, int ptsArmure, ArrayList<Objet> drops)
    {
        super(nom, ptsVie, niveau, att, 0, xpDrop, drops);
        this.ptsArmure = ptsArmure;
        donjonDebloque = niveau + 1;
        this.description = description;
    }

    public String getDescription()
    {
        return description;
    }
    public int getDonjonDebloque() {
        return donjonDebloque;
    }

}

File: KF-Remake/Kingdom-Fall/source/personnages/Boss.cs
namespace personnages;

public class Boss : Ennemi
{
    private readonly int _ptsArmure;
    private readonly int _donjonDebloque;
    private readonly string _description;

    public int DonjonDebloque => _donjonDebloque;
    public string GetDescription => _description;

    public Boss(string nom, string description, int niveau, int ptsVie, int att, int xpDrop, int ptsArmure, List<objets.Objet> drops)
        : base(nom, ptsVie, niveau, att, 0, xpDrop, drops)
    {
        _ptsArmure = ptsArmure;
        _donjonDebloque = niveau + 1;
        _description = description;
    }
}